aten/src/ATen/Utils.h

```diff
@@ -120,4 +120,13 @@ static inline T* get_generator_or_default(const c10::optional<Generator>& gen, c
   return gen.has_value() && gen->defined() ? check_generator<T>(gen) : check_generator<T>(default_gen);
 }
 
+inline void check_size_nonnegative(IntArrayRef size) {
+  for (auto x: size) {
+    TORCH_CHECK(x >= 0, "Trying to create tensor with negative dimension ", x, ": ", size);
+  }
+}
+
+CAFFE2_API
+Tensor empty_cpu(IntArrayRef size, const TensorOptions& options_, c10::optional<c10::MemoryFormat> optional_memory_format);
```

Comment:
I'm a bit skeptical that these should be dumped in the top level `at::` namespace. They don't seem like user-facing functions, and they have extra preconditions for their use which are not necessarily obvious to users.

Reply:
at:: namespace does look like top-level. Should I use existing namespace like at::detail, or just create a new namespace? I think using existing namespace makes sense. Which namespace do you prefer?